sourcemash/frontend/static/js/views/items_view.js

```diff
@@ -14,13 +15,21 @@ Sourcemash.Views.ItemsView = Backbone.View.extend({
     },
 
     markAllAsRead: function() {
+        this.stopListening(this.collection, 'change:unread');
+        this.model.save({unread: false});
```

Comment:
Won't the updateReadStatus get triggered from the for-loop where you do the save on the unread items?